crates/rustc_codegen_spirv/src/codegen_cx/mod.rs

```diff
@@ -233,10 +237,118 @@ impl CodegenArgs {
             "single output or multiple output",
             "[single|multiple]",
         );
+        opts.optflagopt("", "disassemble", "print module to stderr", "");
+        opts.optopt("", "disassemble-fn", "print function to stderr", "NAME");
+        opts.optopt(
+            "",
+            "disassemble-entry",
+            "print entry point to stderr",
+            "NAME",
+        );
+        opts.optflagopt("", "disassemble-globals", "print globals to stderr", "");
         let matches = opts.parse(args)?;
         let module_output_type =
             matches.opt_get_default("module-output", ModuleOutputType::Single)?;
-        Ok(Self { module_output_type })
+        let disassemble = matches.opt_present("disassemble");
+        let disassemble_fn = matches.opt_str("disassemble-fn");
+        let disassemble_entry = matches.opt_str("disassemble-entry");
+        let disassemble_globals = matches.opt_present("disassemble-globals");
```

Comment:
Should some of these flags be incompatible with each other? As in if you pass `--disassemble-fn` and `--disassemble`, you'll get both outputs concatenated.

Reply:
yea probably, but shrug, doesn't matter too much IMO, just adds complexity to an already slightly annoyingly big chunk of code that shouldn't be used anywhere outside of tests

Reply:
Yeah, it probably doesn't matter much, though I do think this has some utility outside of tests. Mainly debugging and sharing snippets, since this does expose `spirv-dis` functionality without needing spirv tools installed.